### PR TITLE
AML Compute users to get Native Notebooks

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #4550

### CODE REVIEW

File: src/client/common/amlContext.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    public get isAmlCompute() {
+        return this.memento.get<boolean>(amlComputeMementoKey, false) || this.isAmlComputeWorkspace();
+    }
+    private isAmlComputeWorkspace() {
+        let isRemoteConnection: boolean = false;
+        // If there is at least 1 folder, we can check if the scheme and authority match a remote connection
+        if (Array.isArray(this.workspace.workspaceFolders) && this.workspace.workspaceFolders.length > 0) {
+            // We are in a remote connection to a CI if the scheme is vscode-remote and the authority contains amlext
+            isRemoteConnection = this.workspace.workspaceFolders.some(
```
Comment (DonJayamanne) [inline src/client/common/amlContext.ts:28]: Logic courtesy of Shantnu (from AML), @SiddhanthUnnithan

File: src/client/telemetry/index.ts [hunk 2]
```diff
@@ -330,6 +330,11 @@ export interface ISharedPropertyMapping {
      * For every DS telemetry we would like to know the type of Notebook Editor used when doing something.
      */
     ['ds_notebookeditor']: undefined | 'old' | 'custom' | 'native';
+    /**
+     * For every DS telemetry we would like to know whether the this is from AML compute or not.
+     * If not in AML compute, then do not send this telemetry.
+     */
+    ['isamlcompute']: 'yes';
```
Comment (DonJayamanne) [inline src/client/telemetry/index.ts:337]: So we can identify this telemetry & exclude from others (as these forced optins into NativeNotebooks could mess up with experiments)

File: src/client/datascience/variablesView/variableViewActivationService.ts [hunk 3]
```diff
@@ -11,12 +11,12 @@ export class VariableViewActivationService implements IExtensionSingleActivation
     constructor(
         @inject(IExtensionContext) private extensionContext: IExtensionContext,
         @inject(IVariableViewProvider) private variableViewProvider: IVariableViewProvider,
-        @inject(IExperimentService) private experimentService: IExperimentService
+        @inject(UseVSCodeNotebookEditorApi) private useVSCodeNotebookEditorApi: boolean
```
Comment (DonJayamanne) [inline src/client/datascience/variablesView/variableViewActivationService.ts:14]: Changes to always use `UseVSCodeNotebookEditorAPI`, to be consistent.

File: src/client/extensionActivation.ts [hunk 4]
```diff
@@ -86,9 +87,11 @@ async function activateLegacy(
     // Load the two data science experiments that we need to register types
     // Await here to keep the register method sync
     const experimentService = serviceContainer.get<IExperimentService>(IExperimentService);
+    const amlCompute = serviceContainer.get<AmlComputeContext>(AmlComputeContext);
     experimentService.logExperiments();
+
     let useVSCodeNotebookAPI =
-        env.appName.includes('Insider') || (await experimentService.inExperiment(Experiments.NativeNotebook));
+        amlCompute.isAmlCompute || (await experimentService.inExperiment(Experiments.NativeNotebook));
```
Comment (rchiodo) [inline src/client/extensionActivation.ts:94]: I think we still need the 'in insiders' check, right? Or is that handled elsewhere now?

Comment (DonJayamanne) [reply src/client/extensionActivation.ts:94]: That's handled in experiment class (was always there, hence removed that check from this place)

File: src/client/common/amlContext.ts (hunk 1 above)
Comment (rchiodo) [reply src/client/common/amlContext.ts:29]: Not super happy with this one off. Feels like we should have the AML extension do this but I don't have a good idea for how we might let them make this decision. 

Opening up our experiment API seems kinda like overkill.

Comment (IanMatthewHuff) [reply src/client/common/amlContext.ts:29]: It does seem a little bit fragile, but aren't we hoping that this all goes away if we can push native notebooks to full usage in a few months?

File: src/client/telemetry/index.ts (hunk 2 above)
Comment (IanMatthewHuff) [reply src/client/telemetry/index.ts:337]: So this part sounds like an unclean mess to me to have to deal with in telemetry. But as long as PMs have signed off on this I guess ok?

Comment (IanMatthewHuff) [reply src/client/telemetry/index.ts:337]: Like, can't someone be an AML compute user and then also get put in the experiment? I guess we can code around in queries, but feels nasty.

**Review by IanMatthewHuff — APPROVED** [review]
I'll confess, I don't love it. The code seems fine, just the design of forcing some in and out outside of the experiment seems messy. But if that's the choice then we do it.

File: src/client/telemetry/index.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/client/telemetry/index.ts:337]: Agreed, and this is why they initially didn't want anyone to be opted into the experiment, but this is required & PMs are ok/happy with this.

**Review by rchiodo — APPROVED** [review]

**Pull request merged**